contracts/protocol/libraries/logic/SupplyLogic.sol

```diff
@@ -504,26 +504,59 @@ library SupplyLogic {
     }
 
     /**
-     * @notice Executes the 'set as collateral' feature. A user can choose to activate or deactivate an asset as
+     * @notice Executes the 'set as collateral' feature. A user can choose to activate an asset as
+     * collateral at any point in time.
+     * @dev Emits the `ReserveUsedAsCollateralEnabled()` event if the asset can be activated as collateral.
+     * @param reservesData The state of all the reserves
+     * @param reservesList The addresses of all the active reserves
+     * @param userConfig The users configuration mapping that track the supplied/borrowed assets
+     * @param asset The address of the asset being configured as collateral
+     * @param tokenIds the ids of the supplied ERC721 token
+     */
+    function executeCollateralizedERC721(
```

Comment:
should we move this out of the supply logic? as it can be executed outside of the supply/borrow context